### PR TITLE
Update doc format of #define macros in iot_common so they render better in VS and doxygen.

### PR DESCRIPTION
**Pull request by ahsonkhan**
Fixes https://github.com/Azure/azure-sdk-for-c/issues/1300

**Before:**
image
image

image
image

**After:**
image
image

image

### CODE REVIEW

File: sdk/inc/azure/iot/az_iot_common.h [hunk 1]
```diff
@@ -108,22 +108,37 @@ typedef enum
...
-  "%24.cid" /**< Used in distributed tracing. More information here: \
-https://docs.microsoft.com/en-us/azure/iot-hub/iot-hub-distributed-tracing. */
-#define AZ_IOT_MESSAGE_PROPERTIES_CONTENT_TYPE \
-  "%24.ct" /**< URL encoded and of the form text%2Fplain or application%2Fjson, etc. */
-#define AZ_IOT_MESSAGE_PROPERTIES_CONTENT_ENCODING "%24.ce" /**< UTF-8, UTF-16, etc. */
-#define AZ_IOT_MESSAGE_PROPERTIES_USER_ID "%24.uid" /**< User ID field. */
-#define AZ_IOT_MESSAGE_PROPERTIES_CREATION_TIME "%24.ctime" /**< Creation time of the message. */
+/// Add unique identification to a message.
+/// @note It can be used with IoT message property API's by wrapping the macro in a
+/// #AZ_SPAN_FROM_STR marco as a parameter, where needed.
```
Comment (danewalton) [inline sdk/inc/azure/iot/az_iot_common.h:113]: this should also render with the 
```c
/**
*
*/
```
right?

File: sdk/inc/azure/iot/az_iot_common.h [hunk 2]
```diff
@@ -108,22 +108,37 @@ typedef enum
...
+#define AZ_IOT_MESSAGE_PROPERTIES_MESSAGE_ID "%24.mid"
+
+/// Used in distributed tracing.
+/// @note More information here:
+/// https://docs.microsoft.com/en-us/azure/iot-hub/iot-hub-distributed-tracing.
+/// @note It can be used with IoT message property API's by wrapping the macro in a
+/// #AZ_SPAN_FROM_STR marco as a parameter, where needed.
```
Comment (danewalton) [inline sdk/inc/azure/iot/az_iot_common.h:120]:
```suggestion
/// #AZ_SPAN_FROM_STR macro as a parameter, where needed.
```

File: sdk/inc/azure/iot/az_iot_common.h [hunk 3]
```diff
@@ -108,22 +108,37 @@ typedef enum
...
+#define AZ_IOT_MESSAGE_PROPERTIES_CORRELATION_ID "%24.cid"
+
+/// URL encoded and of the form text%2Fplain or application%2Fjson, etc.
```
Comment (danewalton) [inline sdk/inc/azure/iot/az_iot_common.h:123]:
```suggestion
/// URL encoded and of the form `text%2Fplain` or `application%2Fjson`, etc.
```

File: sdk/inc/azure/iot/az_iot_common.h [hunk 4]
```diff
@@ -108,22 +108,37 @@ typedef enum
...
+/// @note It can be used with IoT message property API's by wrapping the macro in a
+/// #AZ_SPAN_FROM_STR marco as a parameter, where needed.
```
Comment (danewalton) [inline sdk/inc/azure/iot/az_iot_common.h:125]:
```suggestion
/// #AZ_SPAN_FROM_STR macro as a parameter, where needed.
```

Comment (danewalton) [reply sdk/inc/azure/iot/az_iot_common.h:125]: here and elsewhere

**Review by danewalton — APPROVED** [review]
thanks for doing this @ahsonkhan

File: sdk/inc/azure/iot/az_iot_common.h (hunk 1 above)
Comment (momuno) [reply sdk/inc/azure/iot/az_iot_common.h:113]: That is my understanding as well for block quotes.

Comment (ahsonkhan) [reply sdk/inc/azure/iot/az_iot_common.h:113]: Yes.

**Pull request merged**